lib/src/components/NxFormGroup/__tests__/NxFormGroup.test.tsx

```diff
@@ -5,369 +5,104 @@
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
   it('contains the specified child', function() {
-    const input = <NxStatefulTextInput className="foo" />,
-        component = getShallow({ children: input });
+    const input = <input type="text" data-testid="child" />,
+        view = quickRender({ children: input });
 
-    expect(component).toContainMatchingElement(NxStatefulTextInput);
-    expect(component.find(NxStatefulTextInput)).toHaveClassName('foo');
+    expect(view.getByTestId('child')).toBeInTheDocument();
   });
 
-  describe('nx-label', function() {
-    it('is a <label> child of NxFormGroup', function() {
-      expect(getShallow()).toContainMatchingElement('label.nx-label');
-    });
+  it('includes the label in the text content and as the name of the child form field', function() {
+    const input = <input type="text" />,
+        inputWithId = <input type="text" id="child-id" />,
+        view = quickRender({ children: input }),
+        viewWithId = quickRender({ children: inputWithId });
 
-    it('contains an nx-label__text populated from the label prop', function() {
-      expect(getShallow().find('.nx-label')).toContainMatchingElement('.nx-label__text');
-      expect(getShallow().find('.nx-label__text')).toHaveText('foo');
+    expect(view.container).toHaveTextContent('foo');
+    expect(view.getByRole('textbox')).toHaveAccessibleName('foo');
 
-      expect(getShallow({ label: <span className="foo">bar</span> }).find('.nx-label__text')).toContainReact(
-        <span className="foo">bar</span>
-      );
-    });
-
-    it('has the nx-label--optional class unless the isRequired prop is true', function() {
-      expect(getShallow().find('.nx-label')).toHaveClassName('nx-label--optional');
-      expect(getShallow({ isRequired: undefined }).find('.nx-label')).toHaveClassName('nx-label--optional');
-      expect(getShallow({ isRequired: null }).find('.nx-label')).toHaveClassName('nx-label--optional');
-      expect(getShallow({ isRequired: false }).find('.nx-label')).toHaveClassName('nx-label--optional');
-      expect(getShallow({ isRequired: true }).find('.nx-label')).not.toHaveClassName('nx-label--optional');
-    });
+    expect(viewWithId.container).toHaveTextContent('foo');
+    expect(viewWithId.getByRole('textbox')).toHaveAccessibleName('foo');
   });
 
-  describe('nx-sub-label', function() {
-    it('is not present by default', function() {
-      expect(getShallow()).not.toContainMatchingElement('.nx-sub-label');
-    });
-
-    it('is populated with the sublabel content', function() {
-      expect(getShallow({ sublabel: 'qwerty' })).toContainMatchingElement('.nx-sub-label');
-      expect(getShallow({ sublabel: 'qwerty' }).find('.nx-sub-label')).toHaveText('qwerty');
+  it('does not override the id of the child', function() {
+    const children = <input type="text" id="child-id" />,
+        view = quickRender({ children });
```

Comment:
Should we have the specified id attr to test it does not override the id of the child?

Reply:
Ah I see what it means here.